biz.aQute.resolve/src/biz/aQute/resolve/ProjectResolver.java

```diff
@@ -81,10 +82,14 @@ private String toString(ServiceReference< ? > sr) {
 	private ResolveProcess					resolve		= new ResolveProcess();
 	private Collection<ResolutionCallback>	cbs			= new ArrayList<ResolutionCallback>();
 
-	public ProjectResolver(Project project) {
+	public ProjectResolver(Project project) throws Exception {
 		super(project);
 		getSettings(project);
 		this.project = project;
+
+		Workspace workspace = project.getWorkspace();
+		if (workspace != null)
+			project.addBasicPlugin(new WorkspaceResourcesRepository(workspace));
```

Comment:
This seems wrong. Everytime you create a ProjectResolver, you add a new WorkspaceResourcesRepository plugin to the workspace.